### ISSUE TITLE
save method is resetting existing expire value in case of update

### ISSUE THREAD

**Issue by vhochstein** [redis/redis-om-node#236]
If I update an existing key, which has an expire value set using save method, afterwards expire is reset to -1.

create:
```
const ecareSession = await ecareSessionRepository.save(req.params.sessionid, req.body);
const ttlInSeconds = 1 * 60 * 60; // 1 hours
 await ecareSessionRepository.expire(ecareSession[EntityId], ttlInSeconds);
```

update:
```
const ecareSession = await ecareSessionRepository.save(req.params.sessionid, req.body);
const key = ecareSession[EntityKeyName];
const ttl = await redisUtil.redisClient.ttl(key);
```

ttl is -1, expected round about 3599

```
const ecareSessionSchema = new Schema('ecare-session', {
        data: { type: 'string' }
    },
    {
        dataStructure: 'HASH',
        idStrategy: myRedisIdGenerator,
    }
);
```

**Comment by guyroyse** [comment]
Yep. That's a bug. Might be hard to sort out as internally as the code deletes and re-adds the key using UNLINK and HSET. I did this because calling HSET will overlay the existing fields in the Hash and this leaves existing fields in place. So, if you removed a field in your data, it would not be in your JavaScript object but will still be in Redis.

I'll have to add a check to see if there is a TTL before doing this and reapply it. Or I'll need to look at all the fields that are currently there and the ones in the JavaScript object being written, diff them, and call HDEL on the offending fields.

Frankly, both are a little ugly.

JSON shouldn't have this problem as I don't delete beforehand.

Thanks for the catch!
